addons/website/static/src/snippets/s_process_steps/options.js

```diff
@@ -175,17 +187,35 @@ options.registry.StepsConnector = options.Class.extend({
      * @param {integer} height
      * @returns {string}
      */
-    _getPath(type, width, height) {
+    _getPath(type, width, height, stepHeightDifference) {
         const hHeight = height / 2;
         switch (type) {
             case 's_process_steps_connector_line': {
-                return `M 0 ${hHeight} L ${width} ${hHeight}`;
+                if (stepHeightDifference >= 0) {
+                    return `M 0 ${stepHeightDifference + this.hStepMainElementHeight} L ${width} ${this.hNextStepMainElementHeight}`;
+                } else {
+                    return `M 0 ${this.hStepMainElementHeight} L ${width} ${this.hNextStepMainElementHeight - stepHeightDifference}`;
+                }
             }
             case 's_process_steps_connector_arrow': {
-                return `M ${0.05 * width} ${hHeight} L ${0.95 * width - 6} ${hHeight}`;
+                // When someone plays with the y-axis, it adds the padding in
+                // multiple of 8px. so here we devide it by 8 to calculate the
+                // number of padding steps has been added.
+                const verticalPaddingFactor = Math.abs(stepHeightDifference) / 8;
```

Comment:
You can multiply the verticalPaddingFactor by 1.5 here instead of calculate it 4 times below.